rfcs/text/0008_pulse.md

```diff
@@ -0,0 +1,247 @@
+- The data will be stored in an ES cluster.
+
+#### Endpoints
+
+The following endpoints will send every payload detailed in below encrypted with a similar mechanism to the current telemetry encryption.
+
+##### Authenticate
+
+This Endpoint will be used to retrieve a randomised `deploymentID` and a `token` for the cluster to use in all the subsequent requests. Ideally, it will provide some sort of identifier (like `cluster_uuid` or `license.uuid`) so we can revoke its access to any of the endpoints if explicitly requested ([Blocking telemetry input](https://github.com/elastic/telemetry/pull/221) and [Delete previous telemetry data](https://github.com/elastic/telemetry/issues/209)).
+
+I'd appreciate some insights here to come up with a strong handshake mechanism to avoid stealing identities.
+
+In order to _dereference_ the data, we can store these mappings in a Vault or Secrets provider instead of an index in our ES.
+
+##### Opt-In|Out
+
+Similar to the current telemetry, we want to keep track of when the user opts in or out of telemetry. The implementation can be very similar to the current one. But we recently learned we need to add the origin to know what application has telemetry disabled (Kibana, Beats, Enterprise Search, ...). This makes me wonder if we will ever want to provide a granular option for the user to be able to cherry-pick about what channels are sent and which ones should be disabled.
+
+##### Inject telemetry
+
+In order to minimise the amount of requests, this `POST` should accept bulks of data in the payload (mind the payload size limits if any). It will require authentication based on the `deploymentID` and `token` explained in the [previous endpoint](#authenticate).
+
+The received payload will be pushed to a streaming technology (AWS Firehose, Google Pub/Sub, ...). This way we can maintain a buffer in cases the ingestion of data spikes or we need to stop our ES cluster for any maintenance purposes.
+
+A subscriber to that stream will receive that info a split the payload into smaller documents per channel and index them into their separate indices.
+
+This indexing should also trigger some additional processes like the **generation of instructions** and _special views_ (only if needed, check the point [Access control](#access-control) for more details).
+
+_NB: We might want to consider some sort of piggy-backing to include the instructions in the response. But for the purpose of this RFC, scalability and separation of concerns, I'd rather keep it for future possible improvements._
+
+##### Retrieve instructions
+
+This `GET` endpoint should return the list of instructions generated for that deployment. To control the likely ever-growing list of instructions for each deployment, it will accept a `since` query parameter where the requester can specify the timestamp ever since it was to retrieve the new values.
```

Comment:
Retrieving instructions when there is a mediator in play has the same limitations as the authenticate endpoint.